demos/guide/quic-hq-interop.c

```diff
@@ -607,11 +609,32 @@ static size_t build_request_set(SSL *ssl)
 
         /*
          * We don't strictly have to do this check, but our quic client limits
-         * our max data streams to 100, so we're just batching in groups of 100
+         * our max data streams to 25, so we're just batching in groups of 100
          * for now
+         * NOTE: WE are doing groups of 25 because thats 1/4 of the initial max
+         * stream count that most servers advertise.  This gives the server an
+         * opportunity to send us updated MAX_STREAM frames to extend our stream
+         * allotment, which many servers defer doing.
          */
-        if (poll_count <= 99)
-            new_stream = SSL_new_stream(ssl, 0);
+        if (poll_count <= 25) {
+            for (retry_count = 0; retry_count < 10; retry_count++) {
+                ERR_clear_error();
+                new_stream = SSL_new_stream(ssl, 0);
+                if (new_stream == NULL && (error = ERR_get_error()) != 0) {
+                    if (ERR_GET_REASON(error) == SSL_R_STREAM_COUNT_LIMITED) {
```

Comment:
We generally advise against people checking the error queue for specific reason codes. The code can move about over time, and so the reason codes for any specific error sometimes change. That said there are instances where we document that a specific error code will be raised in certain conditions. By documenting it we formally make it part of the API contract and commit to it. We might want to do that here.

Reply:
documented in SSL_new_stream() pod